src/backend/base/langflow/custom/custom_component/component.py

```diff
@@ -830,4 +837,34 @@ def log(self, message: LoggableType | list[LoggableType], name: str | None = Non
 
     def _append_tool_output(self):
         if next((output for output in self.outputs if output.name == TOOL_OUTPUT_NAME), None) is None:
-            self.outputs.append(Output(name=TOOL_OUTPUT_NAME, display_name="Tool", method="to_toolkit", types=["Tool"]))
+            set_as_tool_input = self._inputs.get("set_as_tool")
+            hidden_value = not set_as_tool_input.value if set_as_tool_input is not None else False
+            self.outputs.append(
+                Output(
+                    name=TOOL_OUTPUT_NAME,
+                    display_name="Tool",
+                    method="to_toolkit",
+                    types=["Tool"],
+                    hidden=hidden_value,
+                )
+            )
+
+    def update_build_config(self, build_config: dotdict, field_value: Any, field_name: str | None = None):
+        return super().update_build_config(build_config, field_value, field_name)
+
+    def _append_tool_toggle_output(self):
+        from langflow.inputs import BoolInput
```

Comment:
You can move this to the top - Langflow imports should always exist

Reply:
Yes, but that creates an issue with the deep copy for recursion. Let me try that again.

Reply:
Error: ImportError: cannot import name 'InputTypes' from partially initialized module 'langflow.inputs.inputs' (most likely due to a circular import) (/Users/edwin.jose/Documents/GitHub/langflow/src/backend/base/langflow/inputs/inputs.py)